discovery_engine_core/core/src/engine.rs

```diff
@@ -1426,6 +1429,39 @@ impl Engine {
     }
 }
 
+fn rank<F, D>(coi: &CoiSystem, user_interests: &UserInterests, default_ord: F, documents: &mut [D])
+where
+    F: Fn(&D, &D) -> Ordering,
+    D: AiDocument,
+{
+    if documents.len() < 2 {
+        return;
+    };
+
+    if let Ok(scores) = coi.score(documents, user_interests) {
+        xayn_discovery_engine_ai::utils::rank(documents, &scores);
+    } else {
+        documents.sort_unstable_by(default_ord);
+    }
+}
+
+fn rank_documents(coi: &CoiSystem, user_interests: &UserInterests, documents: &mut [Document]) {
+    rank(
+        coi,
+        user_interests,
+        |a, b| a.resource.date_published.cmp(&b.resource.date_published),
```

Comment:
❔ I think the should be reversed here, if I haven't overlooked anything.
as in the old rank function:
```rust
            documents
                .sort_unstable_by(|this, other| other.date_published().cmp(&this.date_published()));
```

Reply:
Yes, thanks. I added a test that checks that we are returning the documents in the correct order when we don't rank using the coi.